### DISCUSSION TITLE
Can I use it as window manager?

### DISCUSSION THREAD

**Discussion by Fobo00** [directvt/vtm#257] [Q&A] [answered]
Hey I would just like to know if I could use this instead of a regular window manager like gnome or KDE? I have been actively searching for something like that for a long time and found nothing. If not is there a window manager that could be like that? Like a terminal as the wallpaper and the apps floating above it

**Comment by o-sdn-o** [comment, marked as answer]
Hi!
> I would just like to know if I could use this instead of a regular window manager like gnome or KDE?

The short answer is no.

vtm is currently a text-only application and can only be a text-only application manager (CLI/TUI). In fact, it is a terminal multiplexer.

> is there a window manager that could be like that? Like a terminal as the wallpaper and the apps floating above it

Unfortunately, I don't know of such desktop environments. In all GUI desktop environments that I know of, the terminal is presented as an ordinary graphical application. Perhaps theoretically in some GUI environment it can be placed always in the background expanded to full screen.

**Reply by Fobo00** [reply]
Oh. Sad. Well thanks a lot for answering.

**Reply by Fobo00** [reply]
And sorry, so, it can't run graphical applications right? And if no, will it be the case one day?

**Reply by o-sdn-o** [reply]
Right, it can't for now, but eventually it will fully support GUI applications (as a some kind of GUI-server compatible with win32/X11/Wayland)

**Reply by USLTD** [reply]
@o-sdn-o, please don't add GUI support! Reason why this project is so unique and beautiful is that it brings GUI Windows Manager's powers into Terminals so if you add GUI then this project won't be unique anymore. If you are planning to add APIs like [TermGL](https://github.com/wojciech-graj/TermGL), [pytermgui](https://github.com/bczsalba/pytermgui) or [pyTermTk](https://github.com/ceccopierangiolieugenio/pyTermTk) to emulate GUI APIs in Terminal then alright

**Reply by o-sdn-o** [reply]
At the moment, vtm uses an external terminal as a cell matrix canvas for output. By becoming a GUI application, I mean adding a GUI frontend to vtm that only displays a cell matrix (or a set of matrices as windows). It is needed because vtm is constrained by the external terminal's capabilities. For example, none of the existing terminals supports the grapheme halves output. Also, integration with the operating system's clipboard requires certain GUI functionality. vtm will determine at startup whether it should use its GUI frontend or render in the current terminal.
